### PR TITLE
Add recipe for smart_open

### PR DESCRIPTION
**Pull request by souravsingh**

### CODE REVIEW

**Comment by souravsingh** [comment]
The error arises due to LICENSE file not getting detected. Is there a solution for this?

**Comment by synapticarbors** [comment]
@souravsingh The project does not supply a `MANIFEST.in` file so the `LICENSE` file does not get copied to the pypi package. I would comment out the `license_file` line and add a comment that it is missing and then raise an issue upstream with the package. If/when it makes it into a future release, you can update the feedstock.

**Comment by souravsingh** [comment]
Pinging @pisvorky for his thoughts.

**Comment by tmylk** [comment]
@souravsingh  MANIFEST added in smart_open 1.3.5 released today. Please try again.

**Comment by tmylk** [comment]
Looks great. Thanks @souravsingh !

**Comment by souravsingh** [comment]
The PR is ready for merging

File: recipes/smart_open/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - smart_open
+    - smart_open.tests
+
+about:
+  home: https://github.com/RaRe-Technologies/smart_open
+  license: MIT
+  license_file: LICENSE
+  summary: 'A Python library for efficient streaming of large files'
+
+  # The remaining entries in this section are optional, but recommended
```
Comment (jakirkham) [inline recipes/smart_open/meta.yaml]: Please drop this comment.

File: recipes/smart_open/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - boto >=2.32
+    - bz2file
+    - requests
```
Comment (jakirkham) [inline recipes/smart_open/meta.yaml]: Please drop any of these that are not strictly required for the build. Most likely only `python` and `setuptools`.

File: recipes/smart_open/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+
+test:
+  imports:
+    - smart_open
+    - smart_open.tests
+
+about:
+  home: https://github.com/RaRe-Technologies/smart_open
+  license: MIT
```
Comment (jakirkham) [inline recipes/smart_open/meta.yaml]: Could add `license_family: MIT`.

File: recipes/smart_open/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  description: |
+    simplejson is a is a Python 2 & Python 3 library 
+    for efficient streaming of very large files
+    from/to S3, HDFS, WebHDFS or local (compressed) files. 
+  dev_url: https://github.com/RaRe-Technologies/smart_open
+
+extra:
+  recipe-maintainers:
+    - piskvorky
+    - tmylk
```
Comment (jakirkham) [inline recipes/smart_open/meta.yaml]: Are you ok with being maintainers @piskvorky and @tmylk ?

Also care to add yourself too @souravsingh ?

Comment (piskvorky) [inline recipes/smart_open/meta.yaml]: Not me for sure.

**Comment by tmylk** [comment]
I am ok with being a maintainer together with sourav

File: recipes/smart_open/meta.yaml [hunk 5]
```diff
@@ -38,17 +36,17 @@ test:
...
   home: https://github.com/RaRe-Technologies/smart_open
   license: MIT
+  license_family: 
   license_file: LICENSE
   summary: 'A Python library for efficient streaming of large files'
 
-  # The remaining entries in this section are optional, but recommended
   description: |
-    simplejson is a is a Python 2 & Python 3 library 
+    smart_open is a is a Python 2 & Python 3 library
```
Comment (piskvorky) [inline recipes/smart_open/meta.yaml]: Double "is a is a".

File: recipes/smart_open/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/smart_open/meta.yaml]: Done below.

**Comment by souravsingh** [comment]
@tmylk Sorry for the last commit. It looks like the tests are passing.

**Comment by tmylk** [comment]
Looks good to me.

**Comment by souravsingh** [comment]
I think the PR is ready for merging

**Comment by souravsingh** [comment]
Thanks for merging @synapticarbors

**Pull request merged**